### PR TITLE
Update datafeed dApp with Transparent Forwarder

### PR DESCRIPTION
**Pull request by rajkharvar**

### CODE REVIEW

**Review by yohanelly95 — APPROVED** [review]

File: datafeed-dapp/contracts/Dex.sol [hunk 1]
```diff
@@ -1,147 +1,83 @@
...
+        }(collectionNameHash);
+
+        // * if power is +ve, price = result / 1o^power
+        // * if power is -ve, price = result * 10^power
+        if (collectionPower < 0) {
+            collectionPriceInUSD =
+                collectionResult *
+                uint256(pow(collectionPower));
+        } else {
+            collectionPriceInUSD =
```
Comment (SamAg19) [inline datafeed-dapp/contracts/Dex.sol:48]: multiply with msg.value with result and then divide

File: datafeed-dapp/contracts/Dex.sol [hunk 2]
```diff
@@ -1,147 +1,83 @@
...
         if (val == 0) {
-            return 0;
+            return 1;
         }
-
         int256 result = 1;
 
-        if (val < 0) {
-            result = -1;
+        if (val < -1) {
```
Comment (SamAg19) [inline datafeed-dapp/contracts/Dex.sol:80]: it should be less than 0

**Review by SamAg19 — CHANGES_REQUESTED** [review]

File: datafeed-dapp/contracts/Dex.sol (hunk 1 above)
Comment (rajkharvar) [reply datafeed-dapp/contracts/Dex.sol]: Resolved in [c9f3c35](https://github.com/razor-network/razor-network-examples/pull/5/commits/c9f3c3534e149c4ee881864eededcc5d40e400c7)

File: datafeed-dapp/contracts/Dex.sol (hunk 2 above)
Comment (rajkharvar) [reply datafeed-dapp/contracts/Dex.sol]: Resolved in [c9f3c35](https://github.com/razor-network/razor-network-examples/pull/5/commits/c9f3c3534e149c4ee881864eededcc5d40e400c7)

**Review by SamAg19 — APPROVED** [review]

**Pull request merged**